src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -189,11 +202,22 @@ internal static unsafe byte[] UnsafeDecode(ReadOnlySpan<char> strSpan)
             int decodedLength = strSpan.Length + (4 - mod) % 4;
 
 #if NET6_0_OR_GREATER
+            Decode(strSpan, output, needReplace, decodedLength);
+#else
+            UnsafeDecode(strSpan, output, needReplace, decodedLength);
+#endif
+        }
+
+#if NET6_0_OR_GREATER
+
+        [SkipLocalsInit]
+        private static byte[] Decode(ReadOnlySpan<char> strSpan, bool needReplace, int decodedLength)
```

Comment:
Can these two Decode methods be combined? The main difference is when we create a result array or copy into destination on lines 253 and 311. Besides that a bit of duplicate code.
`byte[] Decode(ReadOnlySpan<char> strSpan, bool needReplace, int decodedLength)`
`void Decode(ReadOnlySpan<char> strSpan, Span<byte> output, bool needReplace, int decodedLength)`